influxdb/test/stream.cc

```diff
@@ -0,0 +1,212 @@
+  void SetUp() override {
+    logging::manager::load();
+    std::thread t{[&] {
+      _server.init();
+      _server.run();
+    }};
+
+    _thread = std::move(t);
+
+    while (!_server.get_init_done());
+  }
+  void TearDown() override {
+    if (_server.get_init_done())
+      _server.stop();
+    _thread.join();
+
+    logging::manager::unload();
+  }
+
+  test_server _server;
+  std::thread _thread;
+};
+
+TEST_F(InfluxDBStream, BadPort) {
+  std::shared_ptr<persistent_cache> cache;
+  storage::metric m1, m2, m3;
+  std::vector<influxdb::column> mcolumns;
+  std::vector<influxdb::column> scolumns;
+
+  ASSERT_THROW(influxdb::stream st("centreon", "pass", "localhost", 4243, "centreon", 3, "host_status", scolumns, "host_metrics", mcolumns, cache), exceptions::msg);
+}
+
+TEST_F(InfluxDBStream, Read) {
+  std::shared_ptr<persistent_cache> cache;
+  storage::metric m1, m2, m3;
+  std::vector<influxdb::column> mcolumns;
+  std::vector<influxdb::column> scolumns;
+  std::shared_ptr<io::data> data;
+  influxdb::stream st("centreon", "pass", "localhost", 4242, "centreon", 3, "host_status", scolumns, "host_metrics", mcolumns, cache);
+
+  ASSERT_THROW(st.read(data, -1), exceptions::msg);
+}
+
+TEST_F(InfluxDBStream, Write) {
+  std::shared_ptr<persistent_cache> cache;
+  storage::metric *m1, *m2, *m3;
+  std::vector<influxdb::column> mcolumns;
+  std::vector<influxdb::column> scolumns;
+  std::shared_ptr<io::data> data;
+  influxdb::stream st("centreon", "pass", "localhost", 4242, "centreon", 3, "host_status", scolumns, "host_metrics", mcolumns, cache);
+
+  m1 = new storage::metric;
+  m2 = new storage::metric;
+  m3 = new storage::metric;
+
+  m1->ctime = 2000llu;
+  m1->interval = 60;
+  m1->is_for_rebuild = true;
+  m1->metric_id = 42u;
+  m1->name = "host1";
+  m1->rrd_len = 42;
+  m1->value = 42.0;
+  m1->value_type = 4;
+  m1->host_id = 1u;
+  m1->service_id = 1u;
+
+  *m2 = *m3 = *m1;
+
+  std::shared_ptr<io::data> d1;
+  std::shared_ptr<io::data> d2;
+  std::shared_ptr<io::data> d3;
+
+  d1.reset(m1);
+  d2.reset(m2);
+  d3.reset(m3);
+  ASSERT_FALSE(st.write(d1));
+  ASSERT_FALSE(st.write(d2));
+  ASSERT_TRUE(st.write(d3));
+}
+
+TEST_F(InfluxDBStream, Flush) {
+  std::shared_ptr<persistent_cache> cache;
+  storage::metric *m1, *m2, *m3;
+  std::vector<influxdb::column> mcolumns;
+  std::vector<influxdb::column> scolumns;
+  std::shared_ptr<io::data> data;
+  influxdb::stream st("centreon", "pass", "localhost", 4242, "centreon", 9, "host_status", scolumns, "host_metrics", mcolumns, cache);
+
+  m1 = new storage::metric;
+  m2 = new storage::metric;
+  m3 = new storage::metric;
+
+  m1->ctime = 2000llu;
+  m1->interval = 60;
+  m1->is_for_rebuild = true;
+  m1->metric_id = 42u;
+  m1->name = "host1";
+  m1->rrd_len = 42;
+  m1->value = 42.0;
+  m1->value_type = 4;
+  m1->host_id = 1u;
+  m1->service_id = 1u;
+
+  *m2 = *m3 = *m1;
+
+  std::shared_ptr<io::data> d1;
+  std::shared_ptr<io::data> d2;
+  std::shared_ptr<io::data> d3;
```

Comment:
I also have added a constructor for metric, the declaration of m1 can be much more shorter and then you can also create d1 with make_shared...